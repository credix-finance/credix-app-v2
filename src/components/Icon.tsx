import React from "react";

type IconNames = "bookmark" | "wallet";

interface IconProps {
	name: IconNames;
	className?: string;
}

export const Icon = ({ name, className }: IconProps) => {
	const defaultClassNames = "w-4 h-4 fill-current";
	className = `${defaultClassNames} ${className ? className : ""}`;

	switch (name) {
		case "bookmark":
			return (
				<svg className={className} viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
					<path d="M2.66699 4.66659C2.66699 2.82564 4.15938 1.33325 6.00033 1.33325H10.0003C11.8412 1.33325 13.3336 2.82564 13.3336 4.66659V12.6594C13.3336 14.3619 11.3421 15.2859 10.0421 14.1865L8.43082 12.8238C8.1823 12.6136 7.81834 12.6136 7.56982 12.8238L5.95848 14.1865C4.65855 15.2859 2.66699 14.3619 2.66699 12.6594V4.66659ZM6.00033 2.66659C4.89576 2.66659 4.00033 3.56202 4.00033 4.66659V12.6594C4.00033 13.2269 4.66418 13.5349 5.09749 13.1684L6.70883 11.8057C7.45439 11.1752 8.54626 11.1752 9.29182 11.8057L10.9031 13.1684C11.3364 13.5349 12.0003 13.2269 12.0003 12.6594V4.66659C12.0003 3.56202 11.1049 2.66659 10.0003 2.66659H6.00033Z" />
				</svg>
			);
		case "wallet":
			return (
				<svg className={className} viewBox="0 0 24 25" xmlns="http://www.w3.org/2000/svg">
					<path d="M17 9.5h5v6h-5a3 3 0 1 1 0-6Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z" />
					<path d="M7.759 4.5h8.482c.805 0 1.47 0 2.01.044.563.046 1.08.145 1.565.392a4 4 0 0 1 1.748 1.748c.247.485.346 1.002.392 1.564.044.541.044 1.206.044 2.01v4.483c0 .805 0 1.47-.044 2.01-.046.563-.145 1.08-.392 1.565a4 4 0 0 1-1.748 1.748c-.485.247-1.002.346-1.564.392-.541.044-1.206.044-2.01.044H7.758c-.805 0-1.47 0-2.01-.044-.563-.046-1.08-.145-1.565-.392a4 4 0 0 1-1.748-1.748c-.247-.485-.346-1.002-.392-1.564C2 16.21 2 15.546 2 14.742v-4.483c0-.805 0-1.47.044-2.01.046-.563.145-1.08.392-1.565a4 4 0 0 1 1.748-1.748c.485-.247 1.002-.346 1.564-.392C6.29 4.5 6.954 4.5 7.758 4.5ZM5.91 6.538c-.438.035-.663.1-.819.18a2 2 0 0 0-.874.874c-.08.156-.145.38-.18.819C4 8.861 4 9.443 4 10.3v4.4c0 .857 0 1.439.038 1.889.035.438.1.663.18.819a2 2 0 0 0 .874.874c.156.08.38.145.819.18.45.037 1.032.038 1.889.038h8.4c.857 0 1.439 0 1.889-.038.438-.035.663-.1.819-.18a2 2 0 0 0 .874-.874c.08-.156.145-.38.18-.819.037-.45.038-1.032.038-1.889v-4.4c0-.857 0-1.439-.038-1.889-.035-.438-.1-.663-.18-.819a2 2 0 0 0-.874-.874c-.156-.08-.38-.145-.819-.18-.45-.037-1.032-.038-1.889-.038H7.8c-.857 0-1.439 0-1.889.038Z" />
				</svg>
			);
		default:
			return null;
	}
};
