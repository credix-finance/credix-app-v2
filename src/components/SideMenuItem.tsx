import React from "react";
import { Icon, IconNames } from "@components/Icon";
import Link from "next/link";

export interface SideMenuItemProps {
	icon: IconNames;
	label: string;
	path: string;
	isActive: boolean;
}

export const SideMenuItem = ({ label, icon, path, isActive }: SideMenuItemProps) => {
	return (
		<Link href={path} key={label}>
			<a className={`h-[60px] flex items-center ${isActive ? "bg-credix-primary" : "bg-darker"}`}>
				<div
					className={`border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center ${
						isActive ? "text-darker font-bold" : "text-credix-primary font-medium"
					}`}
				>
					<Icon name={icon} className="w-[20px] h-[20px]" />
					<span className="pl-[18px] text-base">{label}</span>
				</div>
			</a>
		</Link>
	);
};
