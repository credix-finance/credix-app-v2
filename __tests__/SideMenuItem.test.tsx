import React from "react";
import renderer from "react-test-renderer";
import { IconNames } from "@components/Icon";
import { SideMenuItem } from "@components/SideMenuItem";

const defaultProps = {
	icon: "line-chart" as IconNames,
	label: "Invest",
	path: "/invest",
	isActive: false,
};

test("Side menu item active", () => {
	const activeProps = {
		...defaultProps,
		isActive: true,
	};

	const component = renderer.create(<SideMenuItem {...activeProps} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Side menu item inactive", () => {
	const component = renderer.create(<SideMenuItem {...defaultProps} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
