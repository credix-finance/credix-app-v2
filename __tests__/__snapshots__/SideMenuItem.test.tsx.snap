// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side menu item active 1`] = `
<a
  className="h-[60px] flex items-center bg-credix-primary"
  href="/invest"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center text-darker font-bold"
  >
    <svg
      className="w-4 h-4 fill-current w-[20px] h-[20px]"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="m14.034 7.569-1.225 2.45a2.493 2.493 0 0 1 1.49.745l1.226-2.45a2.494 2.494 0 0 1-1.491-.745Zm-2.612 2.675a2.51 2.51 0 0 0-1.178 1.178L8.578 9.756a2.51 2.51 0 0 0 1.178-1.178l1.666 1.666ZM5.701 9.235l-1.225 2.45a2.493 2.493 0 0 1 1.49.746l1.225-2.45a2.493 2.493 0 0 1-1.49-.746Z"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M16.667 5.833a.833.833 0 1 1-1.667 0 .833.833 0 0 1 1.667 0Zm1.667 0a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm-10 1.667a.833.833 0 1 1-1.667 0 .833.833 0 0 1 1.667 0ZM10 7.5a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm-5 6.667a.833.833 0 1 1-1.666 0 .833.833 0 0 1 1.666 0Zm1.667 0a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm5.833-.834a.833.833 0 1 0 0-1.666.833.833 0 0 0 0 1.666Zm0 1.667a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="pl-[18px] text-base"
    >
      Invest
    </span>
  </div>
</a>
`;

exports[`Side menu item inactive 1`] = `
<a
  className="h-[60px] flex items-center bg-darker"
  href="/invest"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="border-0 h-8 border-l-[3px] border-darker border-solid w-full pl-[23px] flex items-center text-credix-primary font-medium"
  >
    <svg
      className="w-4 h-4 fill-current w-[20px] h-[20px]"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="m14.034 7.569-1.225 2.45a2.493 2.493 0 0 1 1.49.745l1.226-2.45a2.494 2.494 0 0 1-1.491-.745Zm-2.612 2.675a2.51 2.51 0 0 0-1.178 1.178L8.578 9.756a2.51 2.51 0 0 0 1.178-1.178l1.666 1.666ZM5.701 9.235l-1.225 2.45a2.493 2.493 0 0 1 1.49.746l1.225-2.45a2.493 2.493 0 0 1-1.49-.746Z"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M16.667 5.833a.833.833 0 1 1-1.667 0 .833.833 0 0 1 1.667 0Zm1.667 0a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm-10 1.667a.833.833 0 1 1-1.667 0 .833.833 0 0 1 1.667 0ZM10 7.5a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm-5 6.667a.833.833 0 1 1-1.666 0 .833.833 0 0 1 1.666 0Zm1.667 0a2.5 2.5 0 1 1-5 0 2.5 2.5 0 0 1 5 0Zm5.833-.834a.833.833 0 1 0 0-1.666.833.833 0 0 0 0 1.666Zm0 1.667a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="pl-[18px] text-base"
    >
      Invest
    </span>
  </div>
</a>
`;
